Jellyfin.Plugin.Bookshelf/Providers/ComicBookImageProvider.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Model.Drawing;
+using MediaBrowser.Model.Entities;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Jellyfin.Plugin.Bookshelf.Providers.ComicBook
+{
+    /// <summary>
+    /// The ComicBookImageProvider tries find either a image named "cover" or,
+    /// in case that fails, just takes the first image inside the archive,
+    /// hoping that it is the cover.
+    /// </summary>
+    public class ComicBookImageProvider : IDynamicImageProvider
+    {
+        private readonly ILogger<ComicBookImageProvider> _logger;
+
+        private const string CBZ_FILE_EXTENSION = ".cbz";
+
+        public string Name => "Comic book archive";
```

Comment:
I don't think this name is descriptive enough, though I don't have any suggestions..

Additionally, I think it should be Title Case

Reply:
Looking at other providers we have (e.g. for music), I now came up with `Comic Book Zip Archive Cover Extrator`.
This should make it clear that a cover is extracted from a comic book archive and that it's extracted specifically from a Zip archive and not cbr (rar archive) or cb7 (7-zip archive) etc.

It might not be the best name, but it makes clear what the provider does and what it's limitations are. Thought it could be a bit long, but making it shorter will be difficult, especially because the name of the comic format "Comic book archive" is in itself long.